src/client/components/Carousel/Carousel.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useEffect } from 'react';
+import Product from '../Product/Product.component';
+import './CarouselStyle.css';
+
+
+export default function Carousel({ products, show }) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const [length, setLength] = useState(products.length);
```

Comment:
You don't really need a new state to keep track of counts, this may be buggy if length is not updated correctly on time. I suggest sticking to `products.length`